dashboard/views.py

```diff
@@ -154,6 +154,45 @@ def get_queryset(self) -> QuerySet[Achievement]:
 		).order_by('-obtained', '-num_found')
 
 
+class AchievementLeaderboard(
+	LoginRequiredMixin, SuccessMessageMixin, UpdateView[Student, BaseModelForm[Student]]
+):
+	model = Student
+	fields = ('public', )
+	success_url = reverse_lazy('achievement-leaderboard')
+	object: Student
+	template_name = 'dashboard/achievement_leaderboard.html'
+
+	def form_valid(self, form: BaseModelForm[Student]):
+		messages.success(self.request, f"Updated settings for {self.object.user.username}!")
+		form.instance.user = self.request.user
+		return super().form_valid(form)
+
+	def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
+		context = super().get_context_data(**kwargs)
+		context['rows'] = self.get_queryset()
+		return context
+
+	def get_queryset(self) -> List[Dict[str, Any]]:
```

Comment:
You should use a name other than `get_queryset`. This is an `UpdateView`, there is no queryset attached to it, so this will interact with `SingleObjectMixin` in an undesirable way.

Alternatively, you don't really even need this method, since you only use it once in `get_context_data`, you can just fold it in to the above.